Libraries/CO.CDP.MQ.Tests/Outbox/OutboxProcessorTest.cs

```diff
@@ -8,7 +8,7 @@ namespace CO.CDP.MQ.Tests.Outbox;
 public class OutboxProcessorTest
 {
     private readonly Mock<IPublisher> _publisher = new();
-    private readonly Mock<IOutboxMessageRepository> _outbox = new();
+    private readonly InMemoryOutboxMessageRepository _outbox = new();
```

Comment:
I resorted to a hand-crafted test double here, as we're reaching the limitations of Moq. Even if coordination of all the required calls was possible on the outbox repository, it would make tests hard to follow and hard to change.